permitchecklog.go

```diff
@@ -0,0 +1,204 @@
+package authaus
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/IMQS/log"
+	"sync"
+	"time"
+)
+
+// CheckLogEntry represents a single session check log entry
+type CheckLogEntry struct {
+	Timestamp    time.Time `json:"timestamp"`
+	SessionToken string    `json:"session_token"`
+	UserId       UserId    `json:"user_id"`
+}
+
+// CheckUsageTracker manages in-memory storage and periodic flushing of session check logs
+type CheckUsageTracker struct {
+	config           ConfigUsageTracking
+	log              *log.Logger
+	logs             []CheckLogEntry
+	mutex            sync.RWMutex
+	stopChan         chan struct{}
+	doneChan         chan struct{}
+	flushDoneChan    chan struct{}
+	stopOnce         sync.Once
+	flushTicker      *time.Ticker
+	flushing         bool // Track if a flush is in progress
+	db               *sql.DB
+	droppedLogsCount int64
+	memDump          []string // For testing purposes
+}
+
+// NewCheckUsageTracker creates a new usage tracker instance
+func NewCheckUsageTracker(c ConfigUsageTracking, log *log.Logger, db *sql.DB) *CheckUsageTracker {
+	c.SetDefaults()
+	tracker := &CheckUsageTracker{
+		config:        c,
+		log:           log,
+		logs:          make([]CheckLogEntry, 0),
+		stopChan:      make(chan struct{}),
+		doneChan:      make(chan struct{}),
+		flushDoneChan: make(chan struct{}, 1),
+		db:            db, // Database connection will be set later
+	}
+	if tracker.config.Test_MemDump {
+		tracker.memDump = make([]string, 0)
+	}
+	return tracker
+}
+
+func (t *CheckUsageTracker) Initialize(logger *log.Logger) {
+	if logger != nil {
+		t.log = logger
+
+		if t.config.Enabled {
+			t.start()
+		}
+
+		if t.db == nil {
+			t.log.Warn("CheckUsageTracker initialized without a database connection, logs will not be persisted")
+		} else {
+			t.log.Info("CheckUsageTracker initialized with database connection")
+		}
+	}
+}
+
+// LogCheck adds a check request to the in-memory log
+func (t *CheckUsageTracker) LogCheck(sessionToken string, token *Token) {
+	if !t.config.Enabled {
+		return
+	}
+
+	entry := CheckLogEntry{
```

Comment:
Bad bot. We should be handling the pointer to the struct, instead of the struct it self.
```suggestion
	entry := &CheckLogEntry{
```